Refactor Syntex_check class using pylint for checking the syntax (#2)(#15)

riccacocco opened the pull request:
- Integrated features from Compilation_check into syntax_check
- For the syntax_check now we use pylint for syntax validation
- Added documentation for each method

solve #2

riccacocco committed:
test: Implement pytest for the Compilation feature (#11)

Created the class test_compilation used for test the compilation feature
- Added the fixture create_test_files in order to create test file dinamically
- Implemented 4 test cases: one valid python files and 3 non-valid that covers the following area:
 - Invalid python file
 - The file does not exist
 - Missing pylint

riccacocco committed:
Merge branch 'main' into test/compilation

Merged the main in the branch in order to work to the syntax check feature and his tests.

riccacocco committed:
Refactor Syntex_check class using pylint for checking the syntax

- Integrated features from Compilation_check into syntax_check
- For the syntax_check now we use pylint for syntax validation
- Added documentation for each method

riccacocco commented:
need to delete the dedicated test class for compilation as they are no longer suitable for the new class.

eliasfrode commented:
LGTM, maybe add the files to the code directory though?

eliasfrode reviewed (dismissed):
LGTM, maybe add the files to the code directory though?

albinwwoxnerud reviewed (changes requested):
Requested Changes:
- The logic looks good overall.
- However, as @eliasfrode mentioned, please move all logic into code/syntax_check.py for better organization and maintainability.

riccacocco committed:
Merge branch 'main' into test/compilation

riccacocco commented:
> Requested Changes:
> 
> * The logic looks good overall.
> * However, as @eliasfrode mentioned, please move all logic into code/syntax_check.py for better organization and maintainability.

The only files that i see outside code/ are compilation.py, hello_world.py and test_compilation, these files are not usefull because they were used on the first version of the syntax_check, now the logic applied is a bit more complex so these files are useless.
Maybe i can delete them before the merge?

albinwwoxnerud commented:
> > Requested Changes:
> > 
> > * The logic looks good overall.
> > * However, as @eliasfrode mentioned, please move all logic into code/syntax_check.py for better organization and maintainability.
> 
> The only files that i see outside code/ are compilation.py, hello_world.py and test_compilation, these files are not usefull because they were used on the first version of the syntax_check, now the logic applied is a bit more complex so these files are useless. Maybe i can delete them before the merge?

Sound good to me!

riccacocco committed:
Refactor: Removed old unused files

removed the following files because they were working on the previous version of the syntax_version function.
- compilation.py: first version of the syntax_check
- hello_world and test_compilation: old tests files

albinwwoxnerud reviewed (approved):
LGTM
